filament/src/components/TransformManager.cpp

```diff
@@ -240,7 +242,7 @@ void FTransformManager::updateNodeTransform(Instance i) noexcept {
             mAccurateTranslations);
 
     // we apply the parent orientation
-    computeMaterialWorldOrientation(manager[i].materialOrientation, manager[parent].materialOrientation,
+    computeMaterialWorldOrientation(manager[i].materialOrientation, parent.isValid() ? manager[parent].materialOrientation : quatf{0, 0, 0, 1},
```

Comment:
According to the comment in `SingleInstanceComponentManager.h` there is always a dummy element (index=0) what is default initialized. Instead of always checking whether the parent is valid I would just change the dummy element's material orientation to be {0, 0, 0, 1}. This can be achieved by changing the implementation in `quat.h` (this has some unknown affect to the codebase) or update the value in the TransformManager's constructor (localized change).

Reply:
I think we talked about something similar with Robi, I recommended `TransformManager`'s ctor.

Reply:
Yes we talked about it with Bence, I started examining it on yesterday but I did not have the time to finish it yet, will do it by today eod.

Reply:
https://github.com/shapr3d/filament/pull/162/commits/6b05fb0008899ce758a2cd8cfc8e1e4b2f094c46

Reply:
https://github.com/shapr3d/filament/pull/162/commits/86df250ff201e9cacad615ad002f75ae1b478739